tests/RuleSet/RuleSetTest.php

```diff
@@ -168,59 +162,125 @@ public function testResolveRulesWithMissingRuleValue(): void
             ],
-            $ruleSet->getRules()
-        );
-    }
-
-    public function testResolveRulesWithNestedSet(): void
-    {
-        $ruleSet = new RuleSet([
-            '@PHP70Migration' => true,
-            'strict_comparison' => true,
-        ]);
+        ];
 
-        self::assertSameRules(
+        yield 'overriding ruleset with nested set' => [
+            [
+                '@PHP70Migration' => true,
+                'strict_comparison' => true,
+            ],
             [
                 'array_syntax' => true,
                 'strict_comparison' => true,
                 'ternary_to_null_coalescing' => true,
             ],
-            $ruleSet->getRules()
-        );
-    }
-
-    public function testResolveRulesWithDisabledSet(): void
-    {
-        $ruleSet = new RuleSet([
-            '@PHP70Migration' => true,
-            '@PHP54Migration' => false,
-            'strict_comparison' => true,
-        ]);
+        ];
 
-        self::assertSameRules(
+        yield 'overriding ruleset with disabled set' => [
+            [
+                '@PHP70Migration' => true,
+                '@PHP54Migration' => false,
+                'strict_comparison' => true,
+            ],
             [
                 'strict_comparison' => true,
                 'ternary_to_null_coalescing' => true,
             ],
-            $ruleSet->getRules()
-        );
+        ];
+
+        yield 'FQCN as rule name' => [
+            [
+                ArraySyntaxFixer::class => true,
+            ],
+            [
+                'array_syntax' => true,
+            ],
+        ];
+
+        yield 'FQCN as rule name overridden by next entry that uses rule name' => [
+            [
+                ArraySyntaxFixer::class => false,
+                'array_syntax' => true,
+            ],
+            [
+                'array_syntax' => true,
+            ],
+        ];
+
+        yield 'native rule name overridden by next entry with FQCN as rule name (class resolution)' => [
+            [
+                'array_syntax' => false,
+                ArraySyntaxFixer::class => true,
+            ],
+            [
+                'array_syntax' => true,
+            ],
+        ];
+
+        yield 'native rule name overridden by next entry with FQCN as rule name (raw string)' => [
+            [
+                'array_syntax' => false,
+                'PhpCsFixer\Fixer\ArrayNotation\ArraySyntaxFixer' => true,
+            ],
+            [
+                'array_syntax' => true,
+            ],
+        ];
+
+        yield 'same rule configure three times with different approaches for names, but with leading backslash in raw string\'s FQCN' => [
+            [
+                '\PhpCsFixer\Fixer\ArrayNotation\ArraySyntaxFixer' => true,
+                'array_syntax' => false,
+                ArraySyntaxFixer::class => true,
+            ],
+            [
+                'array_syntax' => true,
+            ],
+        ];
```

Comment:
```suggestion
        yield 'same rule configure three times with different approaches for names, but with leading backslash in raw string\'s FQCN' => [
            [
                '\PhpCsFixer\Fixer\ArrayNotation\ArraySyntaxFixer' => true,
                'array_syntax' => ['syntax' => 'long'],
                ArraySyntaxFixer::class => ['syntax' => 'short'],
            ],
            [
                'array_syntax' => ['syntax' => 'short'],
            ],
        ];
```
There are 2 `true` config values, and it is not clear which one ends in the config. Let's make it clear.